### ISSUE TITLE
NullPointer when updating progress bar

### ISSUE THREAD

**Issue by taconaut** [UniversalMediaServer/UniversalMediaServer#551]
When leaving ums idle for a long time, the progress bar in the status tab visually changes from the custom one to a default progress bar. When updating it, a NullPointerException is being thrown. Can't tell what the trigger for this status bar change is.

```
05-05 08:08:48  Exception in thread "Thread-4" java.lang.NullPointerException
05-05 08:08:48  	at net.pms.newgui.GuiUtil$SegmentedProgressBarUI.setValues(GuiUtil.java:285)
05-05 08:08:48  	at net.pms.newgui.StatusTab.updateMemoryUsage(StatusTab.java:457)
05-05 08:08:48  	at net.pms.newgui.StatusTab$3.run(StatusTab.java:465)
05-05 08:08:48  	at java.lang.Thread.run(Unknown Source)
```

**Comment by ExSport** [comment]
Spotted same thing with latest official v5.1.3 on Windows:
```
TRACE 18:00:37.450 [New I/O worker #8] Ignoring self-originating request from /192.168.1.3:51799
TRACE 18:00:37.450 [New I/O worker #6] Sent to socket: Content-Type: text/xml; charset="utf-8"
TRACE 18:00:37.451 [New I/O worker #6] Sent to socket: Cache-Control: no-cache
TRACE 18:00:37.451 [New I/O worker #6] Sent to socket: Expires: 0
TRACE 18:00:37.451 [New I/O worker #6] Sent to socket: Accept-Ranges: bytes
TRACE 18:00:37.451 [New I/O worker #6] Sent to socket: Connection: keep-alive
TRACE 18:00:37.451 [New I/O worker #6] Sent to socket: Server: Windows_Server_2008_R2-amd64-6.1, UPnP/1.0, UMS/5.1.3
TRACE 18:00:37.451 [New I/O worker #6] Sent to socket: Content-Length: 3027
INFO  18:01:22.563 [Thread-4] Exception in thread "Thread-4" java.lang.NullPointerException
INFO  18:01:22.576 [Thread-4] 	at net.pms.newgui.GuiUtil$SegmentedProgressBarUI.setValues(GuiUtil.java:267)
INFO  18:01:22.577 [Thread-4] 	at net.pms.newgui.StatusTab.updateMemoryUsage(StatusTab.java:459)
INFO  18:01:22.578 [Thread-4] 	at net.pms.newgui.StatusTab$3.run(StatusTab.java:467)
INFO  18:01:22.578 [Thread-4] 	at java.lang.Thread.run(Unknown Source)
DEBUG 18:03:37.448 [UPNP-AliveMessageSender] Sending ALIVE...
TRACE 18:03:37.448 [UPNPHelper] Receiving a NOTIFY from [192.168.1.3:56961]
```

**Comment by Nadahar** [comment]
This bug is still very much alive (in 5.2.0 at least). I made a little "debug build" of 5.2.0 that is running on a separate computer, where the only change from standard was the logging line here:
```java
	public void updateMemoryUsage() {
		long max = Runtime.getRuntime().maxMemory() / 1048576;
		long used = (Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()) / 1048576;
		long buf = 0;
		for (RendererConfiguration r : PMS.get().getRenders()) {
			buf += (r.getBuffer());
		}
		LOGGER.debug(String.format("max: %d, used: %d, buf: %d, used-buf: %d",
				(int) max, (int) used, (int) buf, (int) (used-buf)));
		memBarUI.setValues(0, (int) max, (int) (used - buf), (int) buf);
```
I though maybe the error was due to som illegal values in the setValues line, but it doesn't appear so. Neither does it have to do with little memory, as some of my logs show:
```
DEBUG 16:59:42.683 [Thread-4] max: 683, used: 139, buf: 0, used-buf: 139
INFO  16:59:42.683 [Thread-4] Exception in thread "Thread-4" java.lang.NullPointerException
INFO  16:59:42.714 [Thread-4] 	at net.pms.newgui.GuiUtil$SegmentedProgressBarUI.setValues(GuiUtil.java:281)
INFO  16:59:42.714 [Thread-4] 	at net.pms.newgui.StatusTab.updateMemoryUsage(StatusTab.java:471)
INFO  16:59:42.714 [Thread-4] 	at net.pms.newgui.StatusTab$3.run(StatusTab.java:479)
INFO  16:59:42.714 [Thread-4] 	at java.lang.Thread.run(Unknown Source)
```
```
DEBUG 16:59:43.369 [main] max: 683, used: 142, buf: 0, used-buf: 142
ERROR 16:59:43.369 [main] A serious error occurred during PMS init
java.lang.NullPointerException: null
	at net.pms.newgui.GuiUtil$SegmentedProgressBarUI.setValues(GuiUtil.java:281) ~[ums.jar:5.2.0-SNAPSHOT]
	at net.pms.newgui.StatusTab.updateMemoryUsage(StatusTab.java:471) ~[ums.jar:5.2.0-SNAPSHOT]
	at net.pms.newgui.LooksFrame.serverReady(LooksFrame.java:569) ~[ums.jar:5.2.0-SNAPSHOT]
	at net.pms.PMS.init(PMS.java:730) [ums.jar:5.2.0-SNAPSHOT]
	at net.pms.PMS.createInstance(PMS.java:1029) [ums.jar:5.2.0-SNAPSHOT]
	at net.pms.PMS.main(PMS.java:1141) [ums.jar:5.2.0-SNAPSHOT]
```
```
DEBUG 01:18:02.151 [Thread-4] max: 683, used: 63, buf: 0, used-buf: 63
INFO  01:18:02.227 [Thread-4] Exception in thread "Thread-4" java.lang.NullPointerException
INFO  01:18:02.262 [Thread-4] 	at net.pms.newgui.GuiUtil$SegmentedProgressBarUI.setValues(GuiUtil.java:281)
INFO  01:18:02.263 [Thread-4] 	at net.pms.newgui.StatusTab.updateMemoryUsage(StatusTab.java:471)
INFO  01:18:02.272 [Thread-4] 	at net.pms.newgui.StatusTab$3.run(StatusTab.java:479)
INFO  01:18:02.282 [Thread-4] 	at java.lang.Thread.run(Unknown Source)
```
I find it strange that at one instance above it's actually the ```main``` thread that crashes with the bug during init. The bug seems completely random. Can this be a synchronization issue?

**Comment by Nadahar** [comment]
@taconaut @ExSport #663 hopefully fixed this, please try again with latest snapshot.

**Comment by Nadahar** [comment]
#663 Has not fixed the issue, reopening.

**Comment by Nadahar** [comment]
This has hopefully been fixed by #671.
